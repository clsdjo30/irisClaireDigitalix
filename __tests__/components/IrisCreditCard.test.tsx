import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import CreditCard from '../../src/components/IrisCreditCard';

describe('CreditCard', () => {
    const defaultProps = {
        colors: ['#000', '#fff'],
        creditAmount: 100,
        price: 50,
        promoPrice: 40,
        bestDeal: true,
    };

    it('renders correctly', () => {
        const { getByText } = render(<CreditCard {...defaultProps} />);

        expect(getByText('100')).toBeTruthy();
        expect(getByText('Crédit')).toBeTruthy();
        expect(getByText('Best Deal')).toBeTruthy();
        expect(getByText('40€')).toBeTruthy();
        expect(getByText('50€')).toBeTruthy();
    });

    it('does not render Best Deal when bestDeal prop is false', () => {
        const { queryByText } = render(<CreditCard {...defaultProps} bestDeal={false} />);

        expect(queryByText('Best Deal')).toBeNull();
    });

    it('does not render promoPrice when promoPrice prop is not provided', () => {
        const { queryByText } = render(<CreditCard {...defaultProps} promoPrice={undefined} />);

        expect(queryByText('40€')).toBeNull();
    });

    it('matches snapshot', () => {
        const { toJSON } = render(<CreditCard {...defaultProps} />);

        expect(toJSON()).toMatchSnapshot();
    });
});
