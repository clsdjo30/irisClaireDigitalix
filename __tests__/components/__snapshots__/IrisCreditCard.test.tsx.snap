// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard matches snapshot 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4278190080,
        4294967295,
      ],
      "endPoint": [
        1,
        0.5,
      ],
      "locations": [
        0.1,
        0.9,
      ],
      "startPoint": [
        0,
        0.5,
      ],
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#dcdaee",
      "borderColor": "#EEEDF6",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": "15%",
      "marginBottom": 20,
      "width": "90%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#2F2A5A",
            "fontFamily": "mulishBold",
            "fontSize": 20,
          }
        }
      >
        100
      </Text>
      <Text
        style={
          {
            "color": "#2F2A5A",
            "fontFamily": "mulishBold",
            "fontSize": 20,
          }
        }
      >
        Crédit
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#bc9531",
          "borderRadius": 10,
          "left": 120,
          "paddingHorizontal": 20,
          "paddingVertical": 5,
          "position": "absolute",
          "top": -12,
        }
      }
    >
      <Text
        style={
          {
            "color": "#eeedf6",
            "fontFamily": "mulishMedium",
            "fontSize": 14,
          }
        }
      >
        Best Deal
      </Text>
    </View>
    <View
      style={
        {
          "marginLeft": 40,
          "width": "80%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "left": 20,
            "position": "absolute",
            "top": 0,
            "width": 90,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "mulishLight",
              "fontSize": 10,
              "textDecorationLine": "line-through",
            }
          }
        >
          40
          €
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FF7648",
              "height": 20,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text
            style={
              {
                "color": "#eeedf6",
                "fontFamily": "mulishMedium",
                "fontSize": 10,
              }
            }
          >
            -13%
          </Text>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#2F2A5A",
            "fontFamily": "mulishExtraBold",
            "fontSize": 34,
            "marginTop": 20,
          }
        }
      >
        50
        €
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
